src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -238,21 +253,28 @@ private static void UpdateLogWithWorkItemDetails(SarifLog sarifLog, Uri htmlUri,
             }
         }
 
-        private void LogMetricsForFiledWorkItem(SarifLog sarifLog, SarifWorkItemModel sarifWorkItemModel)
+        private void LogMetricsForFiledWorkItem(SarifLog sarifLog, SarifWorkItemModel sarifWorkItemModel, FilingResult filingResult)
         {
-            string guid = Guid.NewGuid().ToString();
+            Debug.Assert(this.FilingResult == FilingResult.None);
+
+            this.FilingResult = filingResult;
+
+            string workItemGuid = Guid.NewGuid().ToString();
+            string logGuid = sarifLog.GetProperty<Guid>("guid").ToString();
             string tags = string.Join(",", sarifWorkItemModel.LabelsOrTags);
 
             var workItemMetrics = new Dictionary<string, object>
                 {
-                    { "correlatingGuid", guid },
+                    { "logGuid", logGuid },
+                    { "workItemGuid", workItemGuid },
                     { "area", sarifWorkItemModel.Area },
                     { "bodyOrDescription", sarifWorkItemModel.BodyOrDescription },
+                    { "filingResult", filingResult.ToString() },
                     { "commentOrDiscussion", sarifWorkItemModel.CommentOrDiscussion },
                     { "htmlUri", sarifWorkItemModel.HtmlUri },
                     { "iteration", sarifWorkItemModel.Iteration },
                     { "labelsOrTags", tags },
-                    { "locationUri", sarifWorkItemModel.LocationUri },
+                    { "locationUri", sarifWorkItemModel.LocationUris },
```

Comment:
>LocationUris [](start = 56, length = 12)

I'm not sure this will serialize as you expect. You might have to join or json serialize them.

Reply:
absolutely right. thanks.

Reply:
we should think about how to raise an exception in the telemetry code for this common mistake.